python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -788,7 +768,8 @@ def _flush(self, _objs=None):
             for obj_id in obj_ids:
                 self.unlock_transaction(obj_id)
 
-            self._lock.release()
+            for obj in objs:
+                obj._releaseParentLock()
```

Comment:
We just have to be very careful to not release a lock if it's already unlocked since that will raise a `ThreadError` or worse, release a lock that someone else is holding because they are using the object. We need to be sure that acquires and releases are always paired and always exception-safe.